pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -89,7 +90,7 @@ while [ "$#" -gt 0 ]; do
         fi
         if [ "$1" != system ]; then
             profile="/nix/var/nix/profiles/system-profiles/$1"
-            mkdir -p -m 0755 "$(dirname "$profile")"
+            mkdir -m 0755 "$(dirname "$profile")"
```

Comment:
I am not sure if this path could be taken twice because if the directory would exist right now the command would error. Or is that the weird case in mkdir where -p is implied?

Reply:
Sorry, I didn't mean to push that change. I got rid of this locally but forgot to force push.

You're right, `-p` also doesn't fail if the directory already exists. We could easily fix that case, but I guess the original question of the intention re. ancestor directories still stands.